query-planner-js/src/QueryPlan.ts

```diff
@@ -46,29 +46,19 @@ export interface FetchNode {
   operationKind: OperationTypeNode;
   operationDocumentNode?: DocumentNode;
   // Optionally describe a number of "rewrites" that query plan executors should apply to the data that is sent as input of this fetch.
-  inputRewrites?: FetchDataInputRewrite[];
```

Comment:
Note to reviewer(s): the context behind the changes of this file is that I always though to ultimately have just `FetchDataRewrite`, but I didn't do a super good job of the first implementation of those rewrites on the execution side, so that before this patch, input and output rewrites use separate code at execution, and I used this separate typing to avoid dead code/ugly assertions in the execution code.

And fwiw, when I implemented this in the router side, I realise there was no point in having that distinction, so the router don't even bother distinguishing between those 2 types. Anyway, this patch cleans up the execution on the gateway side, using the same approach than in the router, and in doing so the distinction between `FetchDataOutputRewrite` and `FetchDataInputRewrite` becomes meaningless, so I figure it's worth getting rid of.

But do note this has no impact on the router since this is just some typing that is changed here: the underlying plans we generate are unchanged.